packages/foundry-sdk-generator/src/generate/betaClient/copyFiles.ts

```diff
@@ -122,37 +83,81 @@ export async function copyFiles(
         fileContents,
         { overwrite: true },
       );
-      for (const importDeclaration of sourceFile.getImportDeclarations()) {
-        const moduleSpecifier = importDeclaration.getModuleSpecifier();
-        const newModuleSpecifier = transformModuleSpecifier(
-          moduleSpecifier.getLiteralValue(),
-          newModulePath,
-        );
-        moduleSpecifier.setLiteralValue(newModuleSpecifier);
-      }
 
-      for (const exportedDecl of sourceFile.getStatements()) {
-        if (Node.isModifierable(exportedDecl)) {
-          exportedDecl.toggleModifier("declare", false);
-        }
+      remapImportsExports(sourceFile, newModulePath, undefined);
+    }
+  }
+  return importSet;
+}
+
+function remapImportsExports(
+  sourceFile: SourceFile,
+  newModulePath: string,
+  importSet: Set<string> | undefined,
+) {
+  for (const importDeclaration of sourceFile.getImportDeclarations()) {
+    const moduleSpecifier = importDeclaration.getModuleSpecifier();
+    const newModuleSpecifier = transformModuleSpecifier(
+      moduleSpecifier.getLiteralValue(),
+      newModulePath,
+    );
+    moduleSpecifier.setLiteralValue(newModuleSpecifier);
+
+    if (importSet && newModuleSpecifier.startsWith("internal")) {
+      for (
+        const importName of importDeclaration.getNamedImports().map(imp =>
+          imp.getName()
+        )
+      ) {
+        importSet.add(`${newModuleSpecifier}:${importName}`);
       }
+    }
+  }
+
+  for (const exportedDecl of sourceFile.getStatements()) {
+    if (Node.isModifierable(exportedDecl)) {
+      exportedDecl.toggleModifier("declare", false);
+    }
+  }
 
-      for (const exportDeclaration of sourceFile.getExportDeclarations()) {
-        const moduleSpecifier = exportDeclaration.getModuleSpecifier();
-
-        if (moduleSpecifier) {
-          const newModuleSpecifier = transformModuleSpecifier(
-            moduleSpecifier.getLiteralValue(),
-            newModulePath,
-          );
-          moduleSpecifier.setLiteralValue(
-            withoutTrailingIndex(newModuleSpecifier),
-          );
+  for (const exportDeclaration of sourceFile.getExportDeclarations()) {
+    const moduleSpecifier = exportDeclaration.getModuleSpecifier();
+    if (moduleSpecifier) {
+      const newModuleSpecifier = transformModuleSpecifier(
+        moduleSpecifier.getLiteralValue(),
+        newModulePath,
+      );
+      moduleSpecifier.setLiteralValue(newModuleSpecifier);
+      if (importSet && newModuleSpecifier.startsWith("internal")) {
+        for (
+          const exportName of exportDeclaration.getNamedExports().map(exp =>
+            exp.getName()
+          )
+        ) {
+          importSet.add(`${newModuleSpecifier}:${exportName}`);
         }
       }
     }
   }
-  return importSet;
+
+  sourceFile.getDescendantsOfKind(SyntaxKind.ImportType).forEach((node) => {
```

Comment:
Here we add the additional check to convert the dynamic imports to their "internal" counter parts